src/utils/AutoId.tsx

```diff
@@ -0,0 +1,51 @@
+/** Based heavily on https://github.com/reach/reach-ui/tree/develop/packages/auto-id */
+import React, { useState, useEffect } from 'react';
+
+let serverHandoffComplete = false;
+let theId = 0;
+const genId = () => {
+  theId += 1;
+  return theId;
+};
```

Comment:
This was originally like this, but linters complained about it ([np-plus-plus](https://eslint.org/docs/rules/no-plusplus) and [no-return-assign](https://eslint.org/docs/rules/no-return-assign)). 
```tsx
const genId = () => ++theId;
```

Current implementation satisfies the linter and was not happy to add ignore comment for linter just for this.